crates/client/rpc/src/tests.rs

```diff
@@ -0,0 +1,125 @@
+            filter_address: None,
+            max_results: None,
+            expected_events: vec![event1.clone(), event2.clone(), event5.clone()],
+            expected_continuation_token: 5,
+        },
+        TestCase {
+            _name: "filter events by address",
+            events: events.clone(),
+            filter_keys: vec![],
+            filter_address: Some(Felt252Wrapper::from_dec_str("2").unwrap()),
+            max_results: None,
+            expected_events: vec![event2.clone()],
+            expected_continuation_token: 5,
+        },
+        TestCase {
+            _name: "filter events by address and keys",
+            events: events.clone(),
+            filter_keys: vec![vec![FieldElement::from(1_u32)], vec![]],
+            filter_address: Some(Felt252Wrapper::from_dec_str("3").unwrap()),
+            max_results: None,
+            expected_events: vec![event5.clone()],
+            expected_continuation_token: 5,
+        },
+        TestCase {
+            _name: "filter events by max results where max results is met",
+            events: events.clone(),
+            filter_keys: vec![vec![FieldElement::from(1_u32)], vec![]],
+            filter_address: None,
+            max_results: Some(1),
+            expected_events: vec![event1.clone()],
+            expected_continuation_token: 1,
+        },
+        TestCase {
+            _name: "filter events by max results where max results is not met",
+            events: events.clone(),
+            filter_keys: vec![vec![FieldElement::from(10_u32)], vec![]],
+            filter_address: None,
+            max_results: Some(1),
+            expected_events: vec![],
+            expected_continuation_token: 5,
+        },
+        TestCase {
+            _name: "filter events where filter_keys.len() < event.keys.len()",
+            events: events.clone(),
+            filter_keys: vec![vec![FieldElement::from(1_u32)]],
+            filter_address: None,
+            max_results: None,
+            expected_events: vec![event1, event2, event4, event5],
+            expected_continuation_token: 5,
+        },
+        TestCase {
+            _name: "filter events where filter_keys.len() > event.keys.len()",
+            events,
+            filter_keys: vec![vec![FieldElement::from(1_u32)], vec![], vec![], vec![]],
+            filter_address: None,
+            max_results: None,
+            expected_events: vec![],
+            expected_continuation_token: 5,
+        },
+    ]
+}
+
+#[rstest]
+#[case::filter_keys(build_test_case()[0].clone())]
+#[case::filter_address(build_test_case()[1].clone())]
+#[case::filters_keys_and_address(build_test_case()[2].clone())]
+#[case::filter_max_results_met(build_test_case()[3].clone())]
+#[case::filter_max_results_not_met(build_test_case()[4].clone())]
+#[case::filter_keys_less_than_actual(build_test_case()[5].clone())]
+#[case::filter_keys_more_than_actual(build_test_case()[6].clone())]
```

Comment:
There is no other edge case you could think of?
Better too much that not enough

Reply:
I thought a little more, I have added two new test cases

- Filter events when there are no filters (return all events)
- Filter events when there are no events (return empty vector)

Apart from these, there are few test cases which I added before, like

- filter_address_chunk_size_met
- filter_address_chunk_size_not_met
- filter_keys_chuck_size_met
- filter_keys_chunk_size_not_met

These felt redundant and I thought it made it challenging to filter the useful cases, so I removed them. However, if you feel we should keep these to protect code changes later, I can add them back.